scripts/globals/events/mog_bonanza.lua

```diff
@@ -0,0 +1,433 @@
+------------------------------------
+-- Mog Bonanza
+-- http://www.playonline.com/ff11us/guide/nomadmogbon/index.html
+-- https://www.bg-wiki.com/ffxi/Category:Mog_Bonanza
+------------------------------------
+require('scripts/globals/npc_util')
+require('scripts/globals/zone')
+------------------------------------
+xi = xi or {}
+xi.events = xi.events or {}
+xi.events.mogBonanza = xi.events.mogBonanza or {}
+xi.events.mogBonanza.data = xi.events.mogBonanza.data or {}
+xi.events.mogBonanza.entities = xi.events.mogBonanza.entities or {}
+
+local localSettings =
+{
+    PEARL_COST            = 200000, -- NOTE: This number is hardcoded in the event
+    MAX_PEARLS            = 1,
+    DISABLE_PRIMEVAL_BREW = 1,      -- 0 will turn on the prank the moogle shows when purchasing
+
+    -- 0x55: New Year's Nomad Mog Bonanza 2021
+    -- 0x5A: 20th Vana'versary Nomad Mog Bonanza
+    -- 0x5C: 21st Vana'versary Nomad Mog Bonanza
+    -- 0x5E: <number>
+    BONANZA_ID = 0x5C,
+
+    -- These are local times, and should be tweaked based on your time zone
+    BUYING_PERIOD_START     = os.time({ year = 2023, month = 5, day = 17, hour = 1, min =  0 }),
+    BUYING_PERIOD_END       = os.time({ year = 2023, month = 6, day = 15, hour = 7, min = 59 }),
+    COLLECTION_PERIOD_START = os.time({ year = 2023, month = 7, day = 11, hour = 1, min =  0 }),
+    COLLECTION_PERIOD_END   = os.time({ year = 2023, month = 7, day = 31, hour = 7, min = 59 }),
+
+    COLLECTION_SERVER_MESSAGE =
+        'Announcing the winning numbers for the 21st Vana\'versary Nomad Mog Bonanza!\n' ..
+        '\n' ..
+        'Rank 3 prize: "7" (last digit)-- 13,298 winners.\n' ..
+        'Rank 2 prize: "71" (last two digits)-- 1,299 winners.\n' ..
+        'Rank 1 prize: "800" (all three digits)-- 62 winners.\n' ..
```

Comment:
Should these be  the 3 values below on lines 48-50 passed into string formatting? Or is that not possible/difficult due to the tabling?

Reply:
Right now the server message is unused while figuring out a way to calculate number of winners.  Though I do agree that this should be the eventual goal (dates, times, winning numbers, and number of winners)